lib/vrl/stdlib/src/parse_csv.rs

```diff
@@ -0,0 +1,130 @@
+use bytes::Buf;
+use csv::ReaderBuilder;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseCsv;
+
+impl Function for ParseCsv {
+    fn identifier(&self) -> &'static str {
+        "parse_csv"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "parse a single CSV formatted row",
+            source: r#"parse_csv!("foo,bar,"foo "", bar"")"#,
+            result: Ok(r#"["foo", "bar", "foo ", bar"]"#),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        let delimiter = arguments.optional("delimiter").unwrap_or(expr!(","));
+        Ok(Box::new(ParseCsvFn { value, delimiter }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "delimiter",
+                kind: kind::BYTES,
+                required: false,
+            },
+        ]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseCsvFn {
+    value: Box<dyn Expression>,
+    delimiter: Box<dyn Expression>,
+}
+
+impl Expression for ParseCsvFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let csv_string = self.value.resolve(ctx)?.try_bytes()?;
+        let delimiter = self.delimiter.resolve(ctx)?.try_bytes()?;
+        if delimiter.len() != 1 {
+            return Err("delimiter must be a single character".into());
+        }
+        let delimiter = delimiter[0];
```

Comment:
Will this work for UTF-8 multi-byte characters? It seems like the implementation here limits us to single byte characters.

Reply:
Yes, indeed, unfortunately this csv-library doesn't support multibyte characters. Since most users probably won't need to override the default delimiter at all or use ascii characters only, do you think we could make it more clear in the documentation and leave it as is? Otherwise we might need to write a csv-parser from scratch(couldn't find another library).

Reply:
Aha, yes, I see that now. I'm ok with just documenting the limitation then 👍